src/main/java/io/vertx/core/shareddata/AsyncMap.java

```diff
@@ -184,6 +201,24 @@ default void replaceIfPresent(K k, V oldValue, V newValue, Handler<AsyncResult<B
    */
   Future<Boolean> replaceIfPresent(K k, V oldValue, V newValue);
 
+  /**
+   * Replace the entry only if it is currently mapped to a specific value
+   *
+   * @param k  the key
+   * @param oldValue  the existing value
+   * @param newValue  the new value
+   * @param ttl  The time to live (in ms) for the entry
+   * @param resultHandler the result handler
+   */
+  default void replaceIfPresent(K k, V oldValue, V newValue, long ttl, Handler<AsyncResult<Boolean>> resultHandler) {
+    replaceIfPresent(k, oldValue, newValue, ttl).onComplete(resultHandler);
+  }
+
+  /**
+   * Same as {@link #replaceIfPresent(K, V, V, long, Handler)} but returns a {@code Future} of the asynchronous result
+   */
+  Future<Boolean> replaceIfPresent(K k, V oldValue, V newValue, long ttl);
```

Comment:
This is a new API method, we should provide a default impl that returns a `Future` failed with `UnsupportedOperationException`.